ts/src/transports/Transport.ts

```diff
@@ -27,40 +29,11 @@ export interface TransportOptions {
   onEnd: (err?: Error) => void;
 }
 
-let selectedTransport: TransportConstructor;
-export function DefaultTransportFactory(transportOptions: TransportOptions): Transport | Error {
-  // The transports provided by DefaultTransportFactory do not support client-streaming
-  if (transportOptions.methodDefinition.requestStream) {
-    return new Error("No transport available for client-streaming (requestStream) method");
-  }
-
-  if (!selectedTransport) {
-    selectedTransport = detectTransport();
-  }
-
-  return selectedTransport(transportOptions);
+export interface TransportFactory {
+  (options: TransportOptions): Transport;
 }
 
-function detectTransport(): TransportConstructor {
-  if (detectFetchSupport()) {
-    return fetchRequest;
-  }
-
-  if (detectMozXHRSupport()) {
-    return mozXhrRequest;
-  }
 
-  if (detectXHRSupport()) {
-    return xhrRequest;
-  }
 
-  if (detectNodeHTTPSupport()) {
-    return httpNodeRequest;
-  }
 
```

Comment:
When you fiew this file it's easy to notice a lot of empty lines - we should probably clean them up.
Have you considered using [prettier](https://github.com/prettier/prettier)?